OrbitGl/TimeGraph.cpp

```diff
@@ -1053,33 +1053,52 @@ void TimeGraph::UpdateMovingTrackSorting() {
   // could be optimized, but this is not worth the effort for the limited number of tracks.
 
   std::shared_ptr<Track> moving_track = nullptr;
+  int moving_track_previous_position = -1;
   for (auto track_it = sorted_filtered_tracks_.begin(); track_it != sorted_filtered_tracks_.end();
        ++track_it) {
     if ((*track_it)->IsMoving()) {
       moving_track = *track_it;
+      moving_track_previous_position = track_it - sorted_filtered_tracks_.begin();
       sorted_filtered_tracks_.erase(track_it);
-      sorted_tracks_.erase(std::find(sorted_tracks_.begin(), sorted_tracks_.end(), moving_track));
       break;
     }
   }
 
   if (moving_track != nullptr) {
-    bool inserted = false;
-
+    int moving_track_current_position = -1;
     for (auto track_it = sorted_filtered_tracks_.begin(); track_it != sorted_filtered_tracks_.end();
          ++track_it) {
       if (moving_track->GetPos()[1] >= (*track_it)->GetPos()[1]) {
-        auto sorted_track_it = std::find(sorted_tracks_.begin(), sorted_tracks_.end(), *track_it);
-        sorted_tracks_.insert(sorted_track_it, moving_track);
         sorted_filtered_tracks_.insert(track_it, moving_track);
-        inserted = true;
+        moving_track_current_position = track_it - sorted_filtered_tracks_.begin();
         break;
       }
     }
 
-    if (!inserted) {
-      sorted_tracks_.push_back(moving_track);
+    if (moving_track_current_position == -1) {
       sorted_filtered_tracks_.push_back(moving_track);
+      moving_track_current_position = static_cast<int>(sorted_filtered_tracks_.size()) - 1;
+    }
+
+    // Now we have to change the position of the moving_track in the non-filtered array
```

Comment:
Nit: The method is getting quite long could be split, maybe into finding the moving track and inserting the moving track at the correct position.

Reply:
better now?

Reply:
Yes, thanks. But another nitpick for the nitpick: `FindMovingTrack` should probably be `FindMovingTrackIndex`, and be protected or private - but visibility is a mess in the TimeGraph class already, so this is at least consistent ;-)

Reply:
haha, done!